### ISSUE TITLE
Data nodes discovery is broken when using then new data tier roles

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/elasticsearch-hadoop#1612]
By default, [ES-Hadoop uses the following](https://www.elastic.co/guide/en/elasticsearch/hadoop/current/configuration.html):
```
es.nodes.wan.only: false
es.nodes.discovery: true
es.nodes.data.only: true
```

Starting with Elasticsearch 7.10, we have node roles such as `data_hot`, `data_warm`, `data_content`, `data_frozen`.
The `data` role is going to be sunset in favor of those roles.

In case ES-Hadoop tries to discover a cluster in search of nodes having the `data` role, it fails.

The problem is at: https://github.com/elastic/elasticsearch-hadoop/blob/2a0eec27275865ab8b9f8022fc59bac454994c43/mr/src/main/java/org/elasticsearch/hadoop/serialization/dto/NodeInfo.java#L61

**Comment by lucabelluccini** [comment]
CC @jbaiera

**Comment by lucabelluccini** [comment]
As a workaround, it's possible to:
1) Set only the data nodes in the `es.nodes` and set `es.nodes.discovery: false`
2) Set only the data nodes in the `es.nodes` and set `es.nodes.wan.only: true`
3) Set `es.nodes.data.only: false`

**Comment by jbaiera** [comment]
Fixed in #1613
